src/asm_files.cpp

```diff
@@ -26,6 +26,8 @@ static vector<T> vector_of(ELFIO::section* sec) {
 }
 
 int create_map_crab(uint32_t map_type, uint32_t key_size, uint32_t value_size, uint32_t max_entries, ebpf_verifier_options_t options) {
+    // BPF_MAP_TYPE_ARRAY_OF_MAPS and
+    // BPF_MAP_TYPE_HASH_OF_MAPS are not yet supported.
     if (map_type == 12 || map_type == 13) {
```

Comment:
FYI, in the future this should probably be a call down through the PAL to something like
platform->is_map_type_supported(map_type)
since the numbers 12 and 13 are Linux platform specific.

Reply:
I can do this in a subsequent PR if you like.

Reply:
Hmm but is the question of support is platform-specific? The support might be related to the choice of abstract domain, and the platform should give some other information about the type of the map (maybe "is_recursive_map" or "value_type" or something). Your suggestion is better than the current state, but maybe this issue is just a hint that this decision shouldn't be here in the first place, but somewhere else.

Reply:
Map types like BPF_MAP_TYPE_CGROUP_ARRAY would only exist on platforms that support cgroups, BPF_MAP_TYPE_XSKMAP would only exist on platforms that support XSK's, etc.  Yes I agree that things like map-in-map can't be supported on any platform until the verifier supports them so we have to have some way to express that.   For now I'm suggesting that it just be hard coded in the platform along with any other relevant constraints.